baseplate/core.py

```diff
@@ -123,6 +124,123 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
         return cls(trace_id, parent_id, span_id, sampled, flags)
 
 
+class AuthenticationContextFactory(object):
```

Comment:
I was afraid of the misunderstanding of what `Authentication` would mean for this.  The token conveys only a basic sense of authentication and the existence of the object does not necessarily correlate to having authentication, it just encapsulates the actual checking of the current authentication.  I don't know if the actual names `AuthenticationContext{,Factory}` make sense but I would prefer that whatever we call it, the wrapper and factory share the same basis rather than have an `[A]Factory` produce not an `[A]`...